driver/src/price_finding/linear_optimization_price_finder.rs

```diff
@@ -232,11 +240,20 @@ fn write_input(input_file: &str, input: &str) -> std::io::Result<()> {
     Ok(())
 }
 
-fn run_solver(input_file: &str) -> Result<(), PriceFindingError> {
+fn run_solver(
+    input_file: &str,
+    optimization_model: OptimizationModel,
+) -> Result<(), PriceFindingError> {
+    let optimization_model_str;
+    match optimization_model {
```

Comment:
This logic could also be a method on OptimizationModel. I'd suggest to have a `to_args() -> &[String]` on the object and implement it for each variant. This way, if we create more sophisticated flavours we just have to implement this method and it's colocated with the data definition.

In any case
```rust
let result = match optimization_model {
  OptimizationModel::MIP => &"mip"
  OptimizationModel::NLP => &"nlp"
}
```

is a bit more idiomatic.